bags_api/src/main/scala/uk/ac/wellcome/platform/storage/bags/api/BagsApi.scala

```diff
@@ -51,59 +88,16 @@ trait BagsApi extends LargeResponses with LookupBag with LookupBagVersions {
         )
 
         get {
-          parameter('version.as[String] ?) {
-            case None => getLatestBag(bagId = bagId)
-            case Some(versionString) =>
-              getBag(bagId = bagId, versionString = versionString)
+          parameter('version.as[String] ?) { maybeVersionString =>
+            withFuture {
+              lookupBagBetter(bagId = bagId, maybeVersionString = maybeVersionString)
```

Comment:
Can we rename this now the previous function is replaced?

Reply:
Oops! Will fix.